daffodil-core/src/main/scala/org/apache/daffodil/dsom/ModelGroup.scala

```diff
@@ -177,7 +180,7 @@ abstract class ModelGroup(index: Int)
    *
    * This also depends on groupMembersDef overrides all being lazy val.
    */
-  final lazy val groupMembers: Seq[Term] =
+  override final lazy val groupMembers: Seq[Term] =
```

Comment:
Giving access to groupMembers probably makes it so the whole walker stuff isn't really necessary. Someone could walk the DSOM entirely with their own implementation rather than relying on all the callbacks and having to store state. Not a big deal I guess, and maybe a benefit?